responses/__init__.py

```diff
@@ -260,14 +260,17 @@ class BaseResponse(object):
 
     stream = False
 
-    def __init__(self, method, url, match_querystring=_unspecified, match=[]):
+    def __init__(self, method, url, match_querystring=_unspecified, match=[], **kwargs):
         self.method = method
         # ensure the url has a default path set if the url is a string
         self.url = _ensure_url_default_path(url)
         self.match_querystring = self._should_match_querystring(match_querystring)
         self.match = match
         self.call_count = 0
 
+        for arg, val in kwargs.items():
+            setattr(self, arg, val)
```

Comment:
Could we set the kwargs to a `request_kwargs` or `extra_args` attribute instead of setting them on the response? Splatting all the kwargs on to the response could limit us in the future as we can't add new properties or methods without potentially breaking userland code.

Reply:
removed and moved to matchers